File: regra2/main_errado.pas
program CopiarArquivo;

uses
  SysUtils;

procedure CopiarArquivo(const origem: string);
var
  diretorio, nomeArquivo, nome, extensao, novoNome, destino: string;
  posExtensao: Integer;
begin
  if FileExists(origem) then
  begin
    diretorio := ExtractFilePath(origem);
    nomeArquivo := ExtractFileName(origem);

    Writeln('Diretório: ', diretorio);
    Writeln('Nome do Arquivo: ', nomeArquivo);

    extensao := ExtractFileExt(nomeArquivo);
    nome := Copy(nomeArquivo, 1, Length(nomeArquivo) - Length(extensao));
  
    novoNome := nome + ' - Cópia' + extensao;
    destino := IncludeTrailingPathDelimiter(diretorio) + novoNome;
    
    if not FileExists(destino) then
    begin
      CopyFile(PChar(origem), PChar(destino), False);
      Writeln('Arquivo copiado para ', destino);
    end
    else
      Writeln('Arquivo ', destino, ' já existe.');
  end
  else
    Writeln('Arquivo ', origem, ' não encontrado.');
end;

begin
  CopiarArquivo('../regra2/bobello.txt');
end.

File: regra2/main_errado.py
import os
import shutil


def copiar_arquivo(origem: str):
    if os.path.exists(origem):
        diretorio, nome_arquivo = os.path.split(origem)
        nome, extensao = os.path.splitext(nome_arquivo)
        novo_nome = f"{nome} - Cópia{extensao}"
        diretorio += "/" + novo_nome
        if not os.path.exists(diretorio):
            shutil.copy(origem, diretorio)
            print(f"Arquivo copiado para {diretorio}")
        else:
            print(f"Arquivo {diretorio} já existe.")
    else:
        print(f"Arquivo {origem} não encontrado.")


if __name__ == "__main__":
    copiar_arquivo("../regra2/bobello.txt")
